src/main/java/org/bytedeco/javacpp/tools/Parser.java

```diff
@@ -938,9 +962,13 @@ Type type(Context context, boolean definition) throws ParserException {
             }
         }
 
-        if (info != null && info.cppTypes != null && info.cppTypes.length > 0) {
+        if (info != null && info.cppTypes != null && info.cppTypes.length > 0 && !info.cppTypes[0].equals(type.cppName)) {
             // use user defined type
             type.cppName = info.cppTypes[0];
```

Comment:
Do we need this? Why did you add this condition in the first place?

Reply:
No we don't. 
I added it when I needed the two lines now in comment below, to avoid an unnecessary call to a new Parser.
The two lines in comment are not strictly needed now, since `type.arguments` is not used further (I think).
If you don't want to uncomment them for now, let's remove the extra condition.

Reply:
Yeah, no, it's alright like that.